File: c_legacy/bench.c
#include <stdio.h>
#include <time.h>
#include "sbuf.h"

int main(void) {
    printf("=== C Legacy Benchmark (1,000,000 Appends) ===\n");

    DynamicBuffer *buf = sbuf_create(16);
    if (buf == NULL) {
        fprintf(stderr, "Allocation failed!\n");
        return 1;
    }

    clock_t start = clock();

    for (int i = 0; i < 1000000; i++) {
        sbuf_append(buf, "CTORM Benchmark String ");
    }

    clock_t end = clock();
    double cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;

    printf("Execution Time : %.4f seconds\n", cpu_time_used);
    printf("Final Capacity : %zu bytes\n", buf->capacity);
    printf("Final Length   : %zu bytes\n", buf->length);

    sbuf_free(buf);
    return 0;
}

File: app/src/bin/bench.rs
use rust_crate::DynamicBuffer;
use std::time::Instant;

fn main() {
    println!("=== Safe Rust Benchmark (1,000,000 Appends) ===");

    let mut buf = DynamicBuffer::with_capacity(16);

    let start = Instant::now();

    for _ in 0..1_000_000 {
        buf.append("CTORM Benchmark String ");
    }

    let duration = start.elapsed();

    println!("Execution Time : {:.4?} seconds", duration.as_secs_f64());
    println!("Final Capacity : {} bytes", buf.capacity());
    println!("Final Length   : {} bytes", buf.len());
}